modules/benchmark/R/calc.benchmark.R

```diff
@@ -48,25 +48,21 @@ calc.benchmark <- function(bm.ensemble, con){
     format_full <- format <- query.format.vars(obvs.id,con) 
 
     # ---- LOAD INPUT DATA ---- #
-    # Right now I'm making the inappropriate assumption that storage type will be 
-    # empty unless it's a time variable. 
-    # This is because I haven't come up for a good way to test that a character string is a date format
 
-    format$vars <- format_full$vars %>% filter(., variable_id %in% db.query(paste(
-      "SELECT variable_id from benchmarks WHERE id IN (", paste(bm.ids, collapse = ","),");"),con)[[1]] | 
-        nchar(storage_type) > 0)  
-    st <- format$vars$storage_type
-    time.row <- which(nchar(st)>1 & substr(st, 1,1) == "%")
+    time.row <- format$time.row
     # vars.used.index is redundant and will be removed when I'm sure it won't break the PDA code. 
-    vars.used.index <- which(nchar(st)==0) 
+    vars.used.index <- which(nchar(format$vars$storage_type)==0) 
     
     obvs <- load.data(data.path, format, start_year, end_year, site, vars.used.index, time.row)
-    dat_vars <- format$vars$pecan_name
+    dat_vars <- format$vars$pecan_name # IF : is this line redundant?
     obvs_full = obvs
     
     # ---- LOAD MODEL DATA ---- #
 
-    model_vars <- format$vars$pecan_name[-time.row]
+    model_vars <- format$vars$pecan_name[-time.row] # IF : what will happen when time.row is NULL? 
+    # For example "AmeriFlux.level2.h.nc" format (38) has time vars year-day-hour listed, but storage type column is empty and it should be
+    # because in load.netcdf we extract the time from netcdf files using the time dimension
+    # we can remove time variables from this format's related variables list or can hardcode "time.row=NULL" in load.x_netcdf function
```

Comment:
First, I don't think this will fail (though it needs to be tested) because here we're removing the time row, so if time.row is null we're not removing anything, but that's fine since time wasn't in the variable list anyway.

Second, since load.data returns all variables in BETY standard names, could we not rely on the column always being named 'time' (or could we not modify read.data to make this true)? If that's the case we could update the time.row variable after load.data by matching (e.g. ```if(is.null(time.row)) time.row = which(colnames(obs) == 'time' ```

Another important thing to consider -- time.row seems ambiguous in what it means. Is it the row in the DATA that contains the time or the row in the RETURNED OBJECT that contains time. I'd assumed the former, which would cause this command to not work anyway. For example, if the format only knew about columns 1, 2 and 38 and col 38 is time, then time.row=38 but format$vars$pecan_name is only length 3.  But looking at the earlier code, it looks like time.row isn't EITHER of these but is the row in the FORMAT that contains the time variable.

Also, shouldn't the model_vars that we load be based on the vars used, not all the vars in the format?

I think the PR is ok, but @bcow should look at these issues